File: 3514-number-of-unique-xor-triplets-ii/3514-number-of-unique-xor-triplets-ii.ts
function uniqueXorTriplets(nums: number[]): number {
    const n = nums.length;
    const MAX = 2048;
    const pairXor: number[] = new Array(MAX).fill(0);
    const tripleXor: number[] = new Array(MAX).fill(0);

    // All pair XORs (including i == j)
    for (let i = 0; i < n; i++) {
        for (let j = i; j < n; j++) {
            pairXor[nums[i] ^ nums[j]] = 1;
        }
    }

    // All triplet XORs
    for (let i = 0; i < n; i++) {
        for (let j = 0; j < MAX; j++) {
            if (pairXor[j] === 1) {
                tripleXor[j ^ nums[i]] = 1;
            }
        }
    }

    // Count distinct
    let cnt = 0;
    for (const val of tripleXor) {
        if (val === 1) cnt++;
    }
    return cnt;
}

File: 3514-number-of-unique-xor-triplets-ii/3514-number-of-unique-xor-triplets-ii.rs
impl Solution {
    pub fn unique_xor_triplets(nums: Vec<i32>) -> i32 {
        const MAX: usize = 2048;
        let mut pair_xor = [0; MAX];
        let mut triple_xor = [0; MAX];

        let n = nums.len();

        // All pair XORs
        for i in 0..n {
            for j in i..n {
                pair_xor[(nums[i] ^ nums[j]) as usize] = 1;
            }
        }

        // All triplet XORs
        for &x in &nums {
            for j in 0..MAX {
                if pair_xor[j] == 1 {
                    triple_xor[(j ^ (x as usize))] = 1;
                }
            }
        }

        // Count distinct
        triple_xor.iter().sum()
    }
}